clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -103,14 +103,46 @@ using namespace std::placeholders;
 namespace {
 
 // Used to check correspondence between allocators and deallocators.
-enum AllocationFamily {
+enum AllocationFamilyKind {
   AF_None,
   AF_Malloc,
   AF_CXXNew,
   AF_CXXNewArray,
   AF_IfNameIndex,
   AF_Alloca,
-  AF_InnerBuffer
+  AF_InnerBuffer,
+  AF_Custom,
+};
+
+class AllocationFamily {
+public:
+  AllocationFamily(AllocationFamilyKind kind,
```

Comment:
I'd suggest marking this `explicit`.

Reply:
Marking with `explicit` would cause compile errors for code like `family == AF_Smth`. We could wrap all `AF_*` with direct constructor call, but, I think, it would be less clean. What do you think?

Reply:
I'd prefer having this explicit.